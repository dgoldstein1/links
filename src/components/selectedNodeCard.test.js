import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import SelectedNodeCard from "./selectedNodeCard";

const mockStore = configureStore([]);

describe("selectedNodeCard", () => {
  describe("matches snapshots", () => {
    let testTable = [
      {
        name: "success",
        state: {
          graph: {
            selectedNode: {
              description: "Test description",
              loading: false,
              node: {
                label: "test"
              }
            }
          }
        }
      },
      {
        name: "no description",
        state: {
          graph: {
            selectedNode: {
              description: undefined,
              loading: false,
              node: {
                label: "test"
              }
            }
          }
        }
      },
      {
        name: "description bigger than word limit",
        state: {
          graph: {
            selectedNode: {
              description:
                "this is a test description longer than the word limit this is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than the word limit",
              loading: false,
              node: {
                label: "test"
              }
            }
          }
        }
      }
    ];

    testTable.forEach(t => {
      it(t.name, () => {
        let store = mockStore(t.state);
        let wrapper = renderer.create(
          <Provider store={store}>
            <SelectedNodeCard />
          </Provider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
