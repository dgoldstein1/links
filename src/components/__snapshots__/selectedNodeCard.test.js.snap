// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectedNodeCard matches snapshots description bigger than word limit 1`] = `
<div
  className="selectedCardContainer"
>
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <div
        className="card fluid"
      >
        <div>
          <h1>
            <a
              href="https://en.wikipedia.org/wiki/test"
            >
              test
            </a>
          </h1>
          <p>
            this is a test description longer than the word limit this is a test description longer than the word limitthis is a test description longer than the word limitthis is a test description longer than t... 
            <a
              href="https://en.wikipedia.org/wiki/test"
            >
              (more)
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`selectedNodeCard matches snapshots no description 1`] = `
<div
  className="selectedCardContainer"
>
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <div
        className="card fluid"
      >
        <div>
          <h1>
            <a
              href="https://en.wikipedia.org/wiki/test"
            >
              test
            </a>
          </h1>
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`selectedNodeCard matches snapshots success 1`] = `
<div
  className="selectedCardContainer"
>
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <div
        className="card fluid"
      >
        <div>
          <h1>
            <a
              href="https://en.wikipedia.org/wiki/test"
            >
              test
            </a>
          </h1>
          <p>
            Test description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
